### PR TITLE
Run Code Coverage Report

### PR DESCRIPTION
**Pull request by stsantilena**
Adds additional stage to the Jenkins pipeline that publishes the HTML code coverage report. 

Is related to: https://github.com/logdna/build-images/pull/54

### CODE REVIEW

**Comment by dkhokhlov** [comment]
wondering - what is the difference in coverage between Linux & windows builds?

**Comment by stsantilena** [comment]
I ran coverage on both Linux and Windows and the percentages come out slightly lower on the Windows side. However, I would expect the Windows numbers to be much lower; which leads me to believe it's calculating the coverage the same way regardless of which tests actually run on a given operating system.

**Comment by dkhokhlov** [comment]
The coverage should depend on actual test runs. And we know that Windows build has noticeably less tests.

File: Makefile [hunk 1]
```diff
@@ -208,6 +208,10 @@ test: unit-test test-journald ## Run unit tests
 unit-test:
 	$(RUST_COMMAND) "--env RUST_BACKTRACE=full --env RUST_LOG=$(RUST_LOG)" "cargo nextest run $(TESTS) --no-fail-fast --nocapture"
 
+.PHONY:unit-code-coverage
```
[the remark this answers is not in the document]
Comment (stsantilena) [reply Makefile]: Good catch! Updated.

[the remark this answers is not in the document]
Comment (dkhokhlov) [reply Makefile]: Will be useful to set a coverage threshold number in rust project file. Such that at the coverage will read it and fail (so build will fail to pass test) if the result coverage falls below it. The idea is that the coverage shall not fall below that threshold, which will be updated up as we improve it.

[the remark this answers is not in the document]
Comment (stsantilena) [reply Makefile]: This is a good idea but will probably require some team discussion on what the thresholds should be and the best way to set them. I think out of scope for this PR, though.

File: common/fs/src/cache/delayed_stream.rs [hunk 2]
```diff
@@ -223,7 +223,7 @@ mod test {
         delay_queue.as_mut().take(50).collect::<Vec<_>>().await;
         let delta = Instant::now() - next;
         let next = Instant::now();
-        assert!(delta < Duration::from_millis(1), "{:?}", delta);
+        assert!(delta < Duration::from_millis(2), "{:?}", delta);
```
Comment (stsantilena) [inline common/fs/src/cache/delayed_stream.rs:226]: Had to adjust this test parameter as it was just above 1 millisecond on the limited, AppVeyor resources.

File: appveyor.yml [hunk 3]
```diff
@@ -21,7 +21,13 @@ install:
   - del rustup-init.exe
   - rustc -vV
   - cargo -vV
+  - rustup component add llvm-tools-preview --toolchain stable-x86_64-pc-windows-msvc
+  - cargo install cargo-llvm-cov
 
 test_script:
   - cargo test
+  - cargo llvm-cov --html
```
Comment (dkhokhlov) [inline appveyor.yml:29]: does this run "test" again? test + cov. if so we should run it only with cov, build time is limited on appveyor.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkhokhlov) [reply Makefile]: The standard procedure for threshold is to set it to the last achieved. The Goal is to not allow it to go down.

File: appveyor.yml (hunk 3 above)
Comment (stsantilena) [reply appveyor.yml:28]: According to the `llvm-cov` [README](https://github.com/taiki-e/cargo-llvm-cov#:~:text=By%20default%2C%20run%20tests%20(via%20cargo%20test)%2C%20and%20print%20the%20coverage%20summary%20to%20stdout.) it does run `cargo test`. Fixed to just run `cargo llvm-cov` to save build time.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (stsantilena) [reply Makefile]: I've created an additional JIRA task for implementing a coverage threshold. I think it'll require additional configuration changes and parsing specific metric output; which is a bit more involved. This PR is just dealing with publishing the HTML reports to our build systems.

**Review by dkhokhlov — APPROVED** [review]
lgtm

**Pull request merged**